qutip/core/coefficient.py

```diff
@@ -630,12 +655,11 @@ def try_parse(code, args, args_ctypes, compile_opt):
         constants = [(f, s, "object") for f, s, _ in constants]
     ncode, variables = use_hinted_type(variables, ncode, args_ctypes)
     if (
-        compile_opt['extra_import']
-        and not compile_opt['extra_import'].isspace()
+        (compile_opt['extra_import']
+        and not compile_opt['extra_import'].isspace())
+        or test_parsed(ncode, variables, constants, args)
```

Comment:
It's difficult to read the indentation for the grouping here.  Also, looks like the `isspace()` call should be more like `if compile_opt['extra_import'].strip()` to account for multiple whitespace?  That could also be done on assignment, rather than on use.

Reply:
`isspace` return True for any numbers and mixes of tabs, newlines and spaces.

Reply:
Ok cool. Still, if you strip it on assignment, then you don't have to remember to test it every time it's used later.